pymc/step_methods/hmc/base_hmc.py

```diff
@@ -166,7 +166,7 @@ def __init__(
         if scaling is None and potential is None:
             mean = floatX(np.zeros(size))
             var = floatX(np.ones(size))
-            potential = QuadPotentialDiagAdapt(size, mean, var, 10, rng=self.rng.spawn(1)[0])
+            potential = QuadPotentialDiagAdapt(size, mean, var, 10, rng=self.rng)
```

Comment:
Why was this spawning a new rng?

Reply:
Because the potential and the step method have to own independent generators. If they were to share the generator, the sampling state will lose that information, and the step method you’d get from the sampling state would produce different draws than the original step method with a shared generator

Reply:
There is one potential per step, so makes sense to have a single source of randomness. The state of the potential should be a subset of the the state of the step.